apps/aecore/src/aec_fork_signalling.erl

```diff
@@ -0,0 +1,350 @@
+%%%============================================================================
+%%% @copyright (C) 2019, Aeternity Anstalt
+%%% @doc
+%%% H1 - height before the the signalling start height;
+%%% HS - signalling start height;
+%%% HE - signalleing end height;
+%%% H2 - fork height;
+%%%
+%%% H1 < HS < HE < H2
+%%%
+%%% The gen_server process (aec_fork_signalling) makes sure that there is a
+%%% fork result at H2 (true, false or pending). In order to compute the fork
+%%% result, it manages asynchronous processes that walk the chain and compute
+%%% how many blocks include a predefined info field between HE - 1 and HS
+%%% (inclusive).
+%%%
+%%% The block at height HE - 1 is the last signalling block, and there must
+%%% be a result computed for it. Once a worker process that computes the
+%%% result is spawned, a 'pending' atom for the given block hash is  stored in
+%%% results map. Once the worker process is done with the computation it will
+%%% send a boolean result and the 'pending' atom is replaced with the boolean
+%%% result.
+%%%
+%%% The blocks with height between HE and FH (inclusive) are treated
```

Comment:
I am surprised on FH/H2 included, and I expect such block cannot be inserted unless outcome at HE - 1 known.

Reply:
Yes, you're right, I'll fix that.